extensions/typescript-language-features/src/tsServer/nodeManager.ts

```diff
@@ -0,0 +1,130 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { TypeScriptServiceConfiguration } from '../configuration/configuration';
+import { setImmediate } from '../utils/async';
+import { Disposable } from '../utils/dispose';
+
+
+const useWorkspaceNodeStorageKey = 'typescript.useWorkspaceNode';
+
+export class NodeVersionManager extends Disposable {
+	private _currentVersion: string | undefined;
```

Comment:
I assume the "version" moniker is left over from copying the tsdk code? Was confused as to whether or not this was checking versions or something until I realized that this was acutally managing the node path.

Reply:
Yeah, do you think simply `NodeManager` would be better?

Reply:
I was imagining `NodePathManager` and `_currentPath`, but I also haven't gone to look for examples that would indicate what's canonical to this repo's style.